src/Repository/PostRepository.php

```diff
@@ -35,21 +35,27 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, Post::class);
     }
 
-    public function findLatest(int $page = 1): Pagerfanta
+    public function findLatest(int $page = 1, string $tag = null): Pagerfanta
     {
-        $query = $this->getEntityManager()
-            ->createQuery('
-                SELECT p, a, t
-                FROM App:Post p
-                JOIN p.author a
-                LEFT JOIN p.tags t
-                WHERE p.publishedAt <= :now
-                ORDER BY p.publishedAt DESC
-            ')
-            ->setParameter('now', new \DateTime())
-        ;
-
-        return $this->createPaginator($query, $page);
+        $qb = $this->createQueryBuilder('p')
+            ->addSelect('a', 't')
+            ->innerJoin('p.author', 'a')
+            ->leftJoin('p.tags', 't')
+            ->where('p.publishedAt <= :now')
+            ->orderBy('p.publishedAt', 'DESC')
+            ->setParameter('now', new \DateTime());
+
+        if (null !== $tag) {
+            $qb->andWhere('p IN (
+                    SELECT p2 
+                    FROM App:Post p2
+                    JOIN p2.tags t2
+                    WHERE t2.name = :tag)
+                ')
+                ->setParameter('tag', $tag);
```

Comment:
@yceruto, I think it's better to use a "secret" DQL function `MEMBER OF` here :). See http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/d)ql-doctrine-query-language.html#dql-select-examples
```php
if (null !== $tag) {
    $qb->andwhere(':tag MEMBER OF p.tags')
       ->setParameter('tag', $tag);
}   
```
To do so, you need to get the tag in the controller:
```php
$tag = $tagRepository->findOneByName($request->query->get('tag'));
$latestPosts = $postRepository->findLatest($page, $tag);
```

Reply:
Excellent idea! That would avoid the subquery, thanks!

Reply:
However, I don't like Doctrine's magic methods `findOneByName`, the IDE can't guess it ;)

Reply:
Just FYI, you can add an annotation to help your IDE like:
```
/**
  * @method Tag findOneByName(string $name)
  */
TagRepository
{
    // ...
}
```

Reply:
Yes, I know :) I meant intuitively, thanks you for share it.